### PR TITLE
fetch size overflow

### PR DESCRIPTION
**Pull request by andrewhao888**
When getting a partial trailing message, consumer's fetch size will double.
However, it is possible to reach max int32, then a kafka broker receives a fetch request with negative fetch size.
Looks like this issue has been checked by Kafka itself with ticket: https://issues.apache.org/jira/browse/KAFKA-7656

Otherwise, we should provide a reasonable default Fetch.Max value

### CODE REVIEW

File: consumer.go [hunk 1]
```diff
@@ -591,6 +592,10 @@ func (child *partitionConsumer) parseResponse(response *FetchResponse) ([]*Consu
 				child.offset++ // skip this one so we can keep processing future messages
 			} else {
 				child.fetchSize *= 2
+				// int32 overflow
```
Comment (varun06) [inline consumer.go]: Can you please change the comment to `check int32 overflow` or remove it?

**Comment by varun06** [comment]
Looks good to me, Do you want to sign CLA and add a test too.

File: consumer.go (hunk 1 above)
Comment (andrewhao888) [reply consumer.go]: sure, let me update the comment

**Comment by andrewhao888** [comment]
Yes, let me add a test case, as well as CLA

**Comment by andrewhao888** [comment]
sorry @varun06, it is too hard to provide a valid test case without mock broker update. A PartialTrailingRecord will lose trailing record after encoding. I just push a test case in fetch_response_test.go and sign CLA. Is that ok to merge?

**Comment by varun06** [comment]
yeah, sounds good.

**Comment by bai** [comment]
Thanks for your contribution, LGTM.

**Pull request merged**